### PR TITLE
Bigpsl writer

### PR DESCRIPTION
**Pull request by mdehoon**
Add a writer to `Bio.Align.bigpsl`.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/SeqFeature.py [hunk 1]
```diff
@@ -1052,6 +1052,18 @@ def __init__(self, start, end, strand=None, ref=None, ref_db=None):
         self.ref = ref
         self.ref_db = ref_db
 
+    def __format__(self, format_spec):
+        if not format_spec:
+            return str(self)
+        if format_spec != "ncbi":
```
[the remark this answers is not in the document]
Comment (mdehoon) [reply Bio/SeqFeature.py:1058]: I guess so, yes. UCSC referred to this as "NCBI style" in their documentation of `pslToBigPsl`. I can change this to `"INSDC"` if that is the appropriate term.

[the remark this answers is not in the document]
Comment (peterjc) [reply Bio/SeqFeature.py]: Yes, see e.g. links in https://github.com/biopython/biopython/blob/master/Bio/SeqIO/InsdcIO.py

In particular https://www.insdc.org/submitting-standards/feature-table/#3.4 (I think they've changed their website, the URL and look is not as I remember).

**Comment by peterjc** [comment]
The location string output is incomplete (as used in GenBank/EMBL/DDBJ files). Essentially we'd want to port the ``_insdc_location_string()`` function in https://github.com/biopython/biopython/blob/master/Bio/SeqIO/InsdcIO.py#L307 to a format method of the location objects.

Or, you could make the Big PSL writer use that?

File: Tests/test_SeqFeature.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
 from Bio.SeqFeature import AfterPosition
 from Bio.SeqFeature import BeforePosition
 from Bio.SeqFeature import BetweenPosition
+from Bio.SeqFeature import Location
```
Comment (peterjc) [inline Tests/test_SeqFeature.py:20]: Looks like this import is now unused?

**Review by peterjc — APPROVED** [review]
I've not looked at the bigpsl specifics, but in terms of impacts to the rest of the code base this looks fine now :)

File: Tests/test_SeqFeature.py (hunk 2 above)
Comment (mdehoon) [reply Tests/test_SeqFeature.py]: Yes; I have removed it.

**Comment by mdehoon** [comment]
> The location string output is incomplete (as used in GenBank/EMBL/DDBJ files). Essentially we'd want to port the `_insdc_location_string()` function in https://github.com/biopython/biopython/blob/master/Bio/SeqIO/InsdcIO.py#L307 to a format method of the location objects.
> 
> Or, you could make the Big PSL writer use that?

I agree that the `_insdc_location_string()` function should be ported to a `__format__` method. That is complicated enough in itself, so for this PR I let the bigPsl writer use `_insdc_location_string()`. Replacing `_insdc_location_string()` by `__format__` can go into a separate PR.

**Comment by peterjc** [comment]
Agreed. I'm sure we discussed adding the feature location to INSDC strings to the SeqFeature methods before, but could not find an open issue.

**Pull request merged**